workers/unity/Assets/Fps/Scripts/Config/FpsEntityTemplates.cs

```diff
@@ -90,16 +82,38 @@ public static EntityTemplate Player(EntityId entityId, string workerId, byte[] a
             const int serverRadius = 150;
             var clientRadius = workerId.Contains(WorkerUtils.MobileClient) ? 60 : 150;
 
-            var serverQuery = InterestQuery.Query(Constraint.RelativeCylinder(serverRadius));
-            var clientQuery = InterestQuery.Query(Constraint.RelativeCylinder(clientRadius));
+            var clientSelfInterest = InterestQuery.Query(Constraint.EntityId(entityId)).FilterResults(new[]
+            {
+                Position.ComponentId, Metadata.ComponentId, OwningWorker.ComponentId,
+                ServerMovement.ComponentId, HealthComponent.ComponentId, GunComponent.ComponentId
+            });
+
+            var clientRangeInterest = InterestQuery.Query(Constraint.RelativeCylinder(clientRadius)).FilterResults(new[]
+            {
+                Position.ComponentId, Metadata.ComponentId, OwningWorker.ComponentId,
+                ServerMovement.ComponentId, ClientRotation.ComponentId, HealthComponent.ComponentId,
+                GunComponent.ComponentId, GunStateComponent.ComponentId, ShootingComponent.ComponentId
+            });
+
+            var serverSelfInterest = InterestQuery.Query(Constraint.EntityId(entityId)).FilterResults(new[]
+            {
+                ClientMovement.ComponentId, ShootingComponent.ComponentId
+            });
+
+            var serverRangeInterest = InterestQuery.Query(Constraint.RelativeCylinder(serverRadius)).FilterResults(new[]
+            {
+                Position.ComponentId, Metadata.ComponentId, OwningWorker.ComponentId,
+                ServerMovement.ComponentId, ClientRotation.ComponentId, HealthComponent.ComponentId,
+                ShootingComponent.ComponentId
+            });
```

Comment:
I'm having a little trouble understanding why we chose these sets for the server interest, a comment could help? :)

Reply:
sure, will add